content/usgs-packages/geoknife_Discovery.Rmd

```diff
@@ -0,0 +1,233 @@
+          rnames=FALSE, align=c("l","l"), col.rgroup = c("none", "#F7F7F7"), 
+          css.cell="padding-bottom: 0.5em; padding-right: 0.5em; padding-top: 0.5em;")
+```
+
+Similar to fabrics where you could not query times without setting the variables, you cannot query attributes of stencils before defining the geoms. Likewise, you cannot query for values of a stencil until you have set the attributes. Attributes give the metadata associated with the stencil and it's geom. Values tell you the individual spatial features available in that attribute of the geom. 
+
+```{r explore-stencil, message=FALSE, warning=FALSE}
+# add a geom to see what values are available
+geom(default_stencil)
+geom(default_stencil) <- "sample:CONUS_states"
+
+# now that geom is set, you can query for available attributes
+query(default_stencil, "attributes")
+attribute(default_stencil) <- "STATE"
+
+# now that attribute is set, you can query for available values
+STATE_values <- query(default_stencil, "values")
+head(STATE_values)
+
+# switch the stencil to see the differences
+ecoreg_stencil <- default_stencil
+geom(ecoreg_stencil) <- "sample:Ecoregions_Level_III"
+query(ecoreg_stencil, "attributes")
+attribute(ecoreg_stencil) <- "LEVEL3_NAM"
+ecoreg_values <- query(ecoreg_stencil, "values")
+head(ecoreg_values)
+```
+
+There are some built-in templates that allow stencils to be defined more specifically. Currently, the package only supports US States, Level III Ecoregions, or HUC8s.
+
+```{r template-stencils, message=FALSE, warning=FALSE}
+# creating geoms from the available templates
+webgeom('state::Wisconsin')
+webgeom('state::Wisconsin,Maine') # multiple states separated by comma
+webgeom('HUC8::09020306,14060009') # multiple HUCs separated by comma
+webgeom('ecoregion::Colorado Plateaus,Driftless Area') # multiple regions separated by comma
+```
+
+## Available webprocesses
+
+The final component to a geojob is the process algorithm used to aggregate the data across the defined stencil. Web process algorithms can be defined by the user, but let's explore the defaults available through GDP.
+
+```{r create-knife, message=FALSE, warning=FALSE}
+# setup a default knife by using webprocess and not supplying any arguments
+default_knife <- webprocess()
+
+# now determine what web processing algorithms are available with the default
+default_algorithms <- query(default_knife, 'algorithms')
+length(default_algorithms)
+head(default_algorithms)
+```
+
+From this list, you can define which algorithm you would like the webprocess component to use. For example, we want our data to be returned as the unweighted average across each geospatial feature, so we need to use "Area Grid Statistics (weighted)".
+
+```{r add-algorithm, message=FALSE, warning=FALSE}
+# algorithm actually has a default of the weighted average
+algorithm(default_knife)
+
+# change the algorithm to unweighted
+algorithm(default_knife) <- default_algorithms['Area Grid Statistics (unweighted)']
```

Comment:
It would probably be best to highlight something other than unweighted (we kind of want to deprecate that one anyway).  I'd recommend OPeNDAP subset.